### PR TITLE
feat(interview): add new feedback page

### PR DESCRIPTION
**Pull request by aaliakseyenka**
#### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 💡 Background and solution
Adds new Feedback page based on the template for interview feedback

Screenshot 2023-08-20 at 15 30 05

### CODE REVIEW

File: client/src/modules/Interviews/pages/InterviewFeedback/getServerSideProps.ts [hunk 1]
```diff
@@ -1,42 +1,145 @@
...
+}: {
+  ctx: GetServerSidePropsContext<ParsedUrlQuery>;
+  token: string | undefined;
+  courseId: number;
+}): Promise<Omit<StageFeedbackProps, keyof CourseOnlyPageProps>> {
+  const studentId = ctx.query.studentId as string | undefined;
+
+  if (!studentId || !ctx.query.interviewId) {
+    throw new Error('No studentId or interviewId');
+  }
```
Comment (valerydluski) [inline client/src/modules/Interviews/pages/InterviewFeedback/getServerSideProps.ts:85]:
```
function validateQueryParams(ctx: GetServerSidePropsContext<ParsedUrlQuery>, params: string[]) {
  for (const param of params) {
    if (!ctx.query[param]) {
      throw new Error(`No ${param}`);
    }
  }
}

  validateQueryParams(ctx, ['studentId', 'interviewId']);

  const studentId = Number(ctx.query.studentId);
  const interviewId = Number(ctx.query.interviewId);

```

File: client/src/modules/Interviews/pages/feedback/feedbackTemplateHandler.ts [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+      }
+
+      // if is rejected, all steps after the current one should be marked as not completed and values should be removed
+      stepMap[step.id] = {
+        values: isRejected ? undefined : step.values,
+        isCompleted: isRejected ? false : step.isCompleted,
+      };
+
+      return stepMap;
+    }, {} as Record<FeedbackStepId, InterviewFeedbackStepData>),
```
Comment (valerydluski) [inline client/src/modules/Interviews/pages/feedback/feedbackTemplateHandler.ts:147]: Maybe it makes sense to extract this into a separate function?
```
function generateFeedbackValues(steps: FeedbackStep[], activeStepIndex: number, newValues: InterviewFeedbackValues, isRejected: boolean): Record<FeedbackStepId, InterviewFeedbackStepData> {
  return steps.reduce((stepMap, step, index) => {
    if (index === activeStepIndex) {
      stepMap[step.id] = {
        isCompleted: true,
        values: newValues,
      };
      return stepMap;
    }

    stepMap[step.id] = {
      values: isRejected ? undefined : step.values,
      isCompleted: isRejected ? false : step.isCompleted,
    };
    return stepMap;
  }, {} as Record<FeedbackStepId, InterviewFeedbackStepData>);
}
```

**Review by valerydluski — APPROVED** [review]

File: client/src/modules/Interviews/pages/InterviewFeedback/getServerSideProps.ts (hunk 1 above)
Comment (aaliakseyenka) [reply client/src/modules/Interviews/pages/InterviewFeedback/getServerSideProps.ts]: thx

File: client/src/modules/Interviews/pages/feedback/feedbackTemplateHandler.ts (hunk 2 above)
Comment (aaliakseyenka) [reply client/src/modules/Interviews/pages/feedback/feedbackTemplateHandler.ts]: sure

**Pull request merged**